modules/planning/lattice/trajectory_generation/trajectory1d_generator.cc

```diff
@@ -28,14 +28,44 @@
 #include "modules/common/log.h"
 #include "modules/planning/common/planning_gflags.h"
 #include "modules/planning/lattice/trajectory1d/constant_deceleration_trajectory1d.h"
-#include "modules/planning/lattice/trajectory1d/standing_still_trajectory1d.h"
 #include "modules/planning/lattice/trajectory1d/lattice_trajectory1d.h"
+#include "modules/planning/lattice/trajectory1d/standing_still_trajectory1d.h"
 #include "modules/planning/math/curve1d/quartic_polynomial_curve1d.h"
-#include "modules/planning/math/curve1d/quintic_polynomial_curve1d.h"
 
 namespace apollo {
 namespace planning {
 
+namespace {
+
+// A common function for trajectory bundles generation with 
+// a given initial and  end conditions.
+typedef std::array<double, 3> InitCondition;
+typedef std::vector<std::pair<std::array<double, 3>, double>> EndConditions;
+template <class PolynomialCurve1dType>
+void GenerateTrajectoryBundlesWithCondition(
+    const InitCondition& init_condition,
+    const EndConditions& end_conditions,
+    std::vector<std::shared_ptr<Curve1d>>* ptr_trajectory_bundle) {
+  CHECK_NOTNULL(ptr_trajectory_bundle);
+
+  for (const auto& end_condition : end_conditions) {
+    // Only the last two elements in the end_condition are useful.
+    std::array<double, 2> end_state;
+    end_state[0] = end_condition.first[1];
+    end_state[1] = end_condition.first[2];
+
```

Comment:
You need to differentiate the trajectory 1d types here and handle them respectively. We have two types of polynomial trajectories: quintic and quartic, and both are used in trajectory generation. Your code here, it is correct for setting up the quartic trajectory, but not for the quintic trajectory.